src/components/GridComponent.js

```diff
@@ -93,12 +103,23 @@ class PathfinderGrid extends Component {
                                 } else if (end.y === rowIndex && end.x === columnIndex) {
                                   type = 'end';
                                 }
-                                return (<Cell passable={cell.passable} key={columnIndex} type={type} handleClick={this.handleCellClick} x={cell.x} y={cell.y} terrain={cell.terrain} />);
+                                let fogVisibility;
+                                if (visibleCells && seenCells) {
```

Comment:
there's a couple of refactoring ideas here **depending on your preference**

1. iirc there's no case where we have only one but not both of `visibleCells` and `seenCells`. we could condense these:
```js
if (visibleCells && !visibleCells.has(cell)) {
  if (seenCells && seenCells.has(cell)) {
    // ...
  } else {
    // ...
  }
}
```
pros: one less layer of indents
cons: longer ifs
(i'd prefer being able to omit the null check but that's up to how you're handling your store)

2. could one-line:
```js
fogVisibility = seenCells.has(cell) ? 'seen' : 'unknown';
```
pros: clearer that it's always set if `!visibleCells.has(cell)`, also you like one-liners
cons: it's one fatter line tho

Reply:
🐇